static/test/javascripts/spec/common/commercial/adblock-messages.spec.js

```diff
@@ -0,0 +1,101 @@
+define([
+    'helpers/injector',
+    'common/utils/storage'
+], function (
+    Injector,
+    storage
+) {
+    var injector = new Injector();
+
+    describe('Adblock messages', function () {
+        var adblockMessage, config, detect, userFeatures;
+
+        beforeEach(function (done) {
+            injector.require([
+                'common/modules/commercial/adblock-messages',
+                'common/utils/config',
+                'common/utils/detect',
+                'common/modules/commercial/user-features'
+            ], function () {
+                adblockMessage = arguments[0];
+                config = arguments[1];
+                detect = arguments[2];
+                userFeatures = arguments[3];
+                done();
+            });
+        });
+
+        describe('If adblock banners and messages should be shown', function () {
+            it('should not show adblock messages for the first time users', function () {
+                storage.local.set('gu.alreadyVisited', 0);
+                config.switches.adblock = true;
+                detect.adblockInUse = function () {
+                    return true;
+                };
+                detect.getBreakpoint = function () {
+                    return 'desktop';
+                };
+
+                expect(adblockMessage.noAdblockMsg()).toBe(true);
+            });
+
+            it('should not show adblock messages when the adblock switch is off', function () {
+                storage.local.set('gu.alreadyVisited', 10);
+                config.switches.adblock = false;
+                detect.adblockInUse = function () {
+                    return true;
+                };
+                detect.getBreakpoint = function () {
+                    return 'desktop';
+                };
+
+                expect(adblockMessage.noAdblockMsg()).toBe(true);
+            });
+
+            it('should not show adblock messages for non adblock users', function () {
```

Comment:
For someone reading these tests, I'm not sure if it's clear what the difference is between `noAdblockMsg()` and `!showAdblockMsg()`